src/dfcx_scrapi/builders/fulfillments.py

```diff
@@ -0,0 +1,305 @@
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+
+class FulfillmentBuilder(BuildersCommon):
+    """Base Class for CX Fulfillment builder."""
+    # TODO: ConditionalCases: def add_conditional_case(self) -> Fulfillment:
+
+    _proto_type = Fulfillment
+    _proto_type_str = "Fulfillment"
+
+
+    def __str__(self) -> str:
+        """String representation of the proto_obj."""
+        try:
+            self._check_proto_obj_attr_exist()
+        except ValueError:
+            return ""
+
+        basic_info_str = self._show_basic_info()
+        resp_msgs_str = self._show_response_messages()
+        params_str = self._show_parameters()
+
+        return (
+            f"Fulfillment Basic Information:\n{'-'*20}\n{basic_info_str}"
+            f"\n\n\nFulfillment ResponseMessages:\n{'-'*20}\n{resp_msgs_str}"
+            f"\n\n\nFulfillment Parameters:\n{'-'*20}\n{params_str}"
+        )
+
+
+    def _show_basic_info(self) -> str:
+        """String representation for the basic information of proto_obj."""
+        self._check_proto_obj_attr_exist()
+
+        partial_resp = self.proto_obj.return_partial_responses
+        return (
+            f"webhook: {self.proto_obj.webhook}"
+            f"\ntag: {self.proto_obj.tag}"
+            f"\nreturn_partial_responses: {partial_resp}"
+        )
+
+
+    def _show_parameters(self) -> str:
+        """String representation for the parameters presets of proto_obj."""
+        self._check_proto_obj_attr_exist()
+
+        return "\n".join([
+            f"{param.parameter}: {param.value if param.value else 'null'}"
+            for param in self.proto_obj.set_parameter_actions
+        ])
+
+
+    def _show_response_messages(self) -> str:
+        """String representation of response messages in proto_obj."""
+        self._check_proto_obj_attr_exist()
+
+        return "\n".join([
+            f"ResponseMessage {i+1}:\n{str(ResponseMessageBuilder(msg))}"
+            for i, msg in enumerate(self.proto_obj.messages)
+        ])
+
+
+    def show_fulfillment(self, mode: str = "whole"):
+        """Show the proto_obj information.
+        Args:
+          mode (str):
+            Specifies what part of the fulfillment to show.
+            Options:
+              ['basic', 'parameters',
+                'messages' or 'response messages', 'whole']
+        """
+        self._check_proto_obj_attr_exist()
+
+        if mode == "basic":
+            print(self._show_basic_info())
+        elif mode == "parameters":
+            print(self._show_parameters())
+        elif mode in ["messages", "response messages"]:
+            print(self._show_response_messages())
+        elif mode == "whole":
+            print(self)
+        else:
+            raise ValueError(
+                "mode should be in"
+                "['basic', 'parameters',"
+                " 'messages' or 'response messages', 'whole']"
+            )
+
+
+    def create_new_proto_obj(
```

Comment:
@MRyderOC there's an edge case here that needs to be adjusted.

If you call `FulfillmentBuilder.create_new_proto_obj` and leave all of the default keyword args as-is, no `self.proto_obj` is created.

I see this as perhaps a more common need to create an "empty" Fulfillment before loading it up using `add_response_message`, etc.

For example:
- I want to create an Empty Fulfillment
- I DO NOT want to use a webhook
- I will call `create_new_proto_obj` but I have no need to add any of the webhook info because I don't need a webhook

In this scenario, the Fulfillment object is not created.

All other scenarios this method works great!

Reply:
Great catch! It took me a while to find the bug and it was in the `__init__` and `_check_proto_obj_attr_exist` of BuildersCommon. If you check an empty fulfillment against if statement it behaves that there is no fulfillment.
```
if not some_empty_fulfillment:
    print("something")
# it would print something
```
The problems are here:
1. https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/ec49fc4a7fd97ba5c3c7929821e288f2a1429df7/src/dfcx_scrapi/builders/builders_common.py#L40-L41
2. https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/ec49fc4a7fd97ba5c3c7929821e288f2a1429df7/src/dfcx_scrapi/builders/builders_common.py#L46-L50


I had to check these conditions to make sure `obj` or `self.proto_obj` is not None (which makes more sense!)

commit link: https://github.com/GoogleCloudPlatform/dfcx-scrapi/commit/496c2cd6299e61808945d631b01ef59ed8d2b389